src/Razor/src/Microsoft.VisualStudio.DevKit.Razor/DevKitTelemetryReporter.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.VisualStudio.Telemetry;
+
+#if DEBUG
+using System.Linq;
+#endif
+
+namespace Microsoft.VisualStudio.DevKit.Razor;
+
+[Shared]
+[Export(typeof(IDevKitTelemetryReporter))]
+internal sealed class DevKitTelemetryReporter : IDevKitTelemetryReporter
```

Comment:
cc @ryzngard - a significant amount of this logic is borrowed from our existing TelemetryReporter. Is there anything we need to do explicitly differently in VS Code besides utilizing InitializeSession?

Reply:
Yea, other than initializing this should be the exact same.